code/API_definitions/connected-network-type.yaml

```diff
@@ -0,0 +1,416 @@
+                Connected-to-4GLTE:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: 4GLTE
+                Connected-To-5GNSA:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: 5GNSA
+                Connected-To-Non3gpp4g:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: NON3GPP4G
+                Not-Connected:
+                  value:
+                    lastStatusTime: "2024-02-20T10:41:38.657Z"
+                    connectedNetworkType: NOT_CONNECTED
+        "400":
+          $ref: "#/components/responses/Generic400"
+        "401":
+          $ref: "#/components/responses/Generic401"
+        "403":
+          $ref: "#/components/responses/ConnectedNetworkTypePermissionDenied403"
+        "404":
+          $ref: "#/components/responses/ConnectedStatusNotFound404"
+        "422":
+          $ref: "#/components/responses/ConnectedStatusUnprocessableEntity422"
+        "500":
+          $ref: "#/components/responses/Generic500"
+        "503":
+          $ref: "#/components/responses/Generic503"
+components:
+  securitySchemes:
+    openId:
+      type: openIdConnect
+      openIdConnectUrl: https://example.com/.well-known/openid-configuration
+  parameters:
+    x-correlator:
+      name: x-correlator
+      in: header
+      description: Correlation id for the different services
+      schema:
+        type: string
+  headers:
+    x-correlator:
+      description: Correlation id for the different services
+      schema:
+        type: string
+  schemas:
+    LastStatusTime:
+      description: Last time that the associated connected Network Type was updated
+      type: string
+      format: date-time
+      example: "2024-02-20T10:41:38.657Z"
+
+    ConnectedNetworkTypeResponse:
+      type: object
+      required:
+        - connectedNetworkType
+      properties:
+        lastStatusTime:
+          $ref: "#/components/schemas/LastStatusTime"
+        connectedNetworkType:
+          $ref: "#/components/schemas/ConnectedNetworkType"
+
+    ConnectedNetworkType:
+      description: |
+        * NOT_CONNECTED: The device is not connected to network
+        * 2G: 2nd Generation Mobile Communication Technology
+        * 3G: 3rd Generation Mobile Communication Technology
+        * 4GLTE: 4th Generation Mobile Communication Technology aka LTE
+        * 5GNSA:4th Generation Mobile Communication Technology aka LTE with device connected 5G New Radio (NR)
+        * 5GSA: 5th Generation Mobile Communication Technology
+        * NON3GPP4G: Device connected via non-3GPP Radio Access Technology but connected to 4G Network (e.g. WiFi Calling)
+        * NON3GPP5G: Device connected via non-3GPP Radio Access Technology but connected to 5G Network (e.g. WiFi Calling)
+      type: string
```

Comment:
```suggestion
        * NTN4G: Device connected over 4G via 3GPP non-terrestrial network 
        * NTN5G: Device connected over 5G via 3GPP non-terrestrial network 
      type: string
```

Reply:
Suggestion above is to add non-terrestrial network (satellite) access, because it will incur increased propagation delay and reduced throughput vs. terrestrial 4G/5G

Reply:
> Suggestion above is to add non-terrestrial network (satellite) access

The problem with adding this option is that almost certainly nobody will support it. Why not? Because the device will almost certainly be roaming on such networks. And I don't see any API provider implementing a solution to determine what access technology a roaming device is using.

Non-terrestrial networks would be better detected using the roaming API, as they should be using a non-geographic PLMN.

Reply:
> ... where satellite is filling coverage gaps to regular UEs

But you are still referring to a "direct to mobile" service using a 3GPP standard (4G or 5G)? Who is proposing to offer such a service using their normal terrestrial PLMN ID?

Reply:
> Who is proposing to offer such a service using their normal terrestrial PLMN ID?

Too early to say, but with reports of dozens of markets considering satellite-coverage deployment, then it makes sense to me to cover the valid 'same PLMN ID' option in the API  (either as an enum value or with the optional modifier I mention above).

Reply:
Good discussion and thoughts, but it will be best to start with a simpler list of connection technology list.  As mentioned above, in future the list can be modified.